### PR TITLE
HAWQ-716. Support code coverage build and report in hawq Makefile

### PR DESCRIPTION
**Pull request by huor**

### CODE REVIEW

**Comment by paul-guo-** [comment]
1) You might want to avoid using the absolute path of lcov, etc. (users could install them anywhere).
2)  Does the code check whether lcov packages have been installed at the configure stage?

**Comment by huor** [comment]
@paul-guo-, please find my comments below:
1. it now uses lcov, gcov, genhtml directory instead of using the absolute path, assuming they are system search path
2. lcov, gcov, genhtml are checked in configure already

**Comment by paul-guo-** [comment]
+1

File: src/bin/gpfdist/Makefile [hunk 1]
```diff
@@ -22,7 +22,8 @@ include $(top_builddir)/src/Makefile.global
 
 LIBS := -levent -lyaml -lz -lbz2 -lssl -lcrypto
 ifeq ($(BUILD_TYPE), gcov)
-LIBS := -lgcov $(LIBS)
+LIBS   := -lgcov $(LIBS)
```
Comment (xunzhang) [inline src/bin/gpfdist/Makefile]: indent

**Comment by xunzhang** [comment]
LGTM

File: src/bin/gpfdist/Makefile (hunk 1 above)
Comment (huor) [inline src/bin/gpfdist/Makefile]: The indent of LIBS is correct since it is consistent with CFLAGS in next line.

**Pull request closed without merging**